src/components/actions/ActionBarReducers.ts

```diff
@@ -0,0 +1,53 @@
+import { IActionOptions } from './Action';
+import { ActionBarActions } from './ActionBarActions';
+import { IReduxAction } from '../../utils/ReduxUtils';
+import { IReduxActionPayload } from '../../ReactVapor';
+import * as _ from 'underscore';
+
+export interface IActionBarState {
+  id: string;
+  actions: IActionOptions[];
+}
+
+export const actionBarInitialState: IActionBarState = { id: undefined, actions: [] };
+export const actionBarsInitialState: IActionBarState[] = [];
+
+export const actionBarReducer = (state: IActionBarState = actionBarInitialState, action: IReduxAction<IReduxActionPayload>): IActionBarState => {
+  switch (action.type) {
+    case ActionBarActions.addActions:
+      if (state.id !== action.payload.id) {
+        return state;
+      }
+      return {
+        id: state.id,
+        actions: action.payload.actions
+      };
+    case ActionBarActions.add:
+      return {
+        id: action.payload.id,
+        actions: []
+      };
+    default:
+      return state;
+  }
+};
+
+export const actionBarsReducer = (state: IActionBarState[] = actionBarsInitialState, action: IReduxAction<IReduxActionPayload>): IActionBarState[] => {
```

Comment:
I renamed in the next one (the one I was working on when we talked about it), I will do so for this one as well.